File: src/main/java/com/lcaohoanq/nocket/domain/post/PostListener.java
package com.lcaohoanq.nocket.domain.post;

import com.lcaohoanq.nocket.domain.cache.IPostRedisService;
import jakarta.persistence.PostPersist;
import jakarta.persistence.PostRemove;
import jakarta.persistence.PostUpdate;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreRemove;
import jakarta.persistence.PreUpdate;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

@Component
@Slf4j
@RequiredArgsConstructor
public class PostListener {

    private final IPostRedisService postRedisService;
    
    @PrePersist
    public void prePersist(Post post) {
        log.info("prePersist");
    }

    @PostPersist //save = persis
    public void postPersist(Post post) {
        // Update Redis cache
        log.info("postPersist");
        postRedisService.clear();
    }

    @PreUpdate
    public void preUpdate(Post post) {
        //ApplicationEventPublisher.instance().publishEvent(event);
        log.info("preUpdate");
    }

    @PostUpdate
    public void postUpdate(Post post) {
        // Update Redis cache
        log.info("postUpdate");
        postRedisService.clear();
    }

    @PreRemove
    public void preRemove(Post post) {
        //ApplicationEventPublisher.instance().publishEvent(event);
        log.info("preRemove");
    }

    @PostRemove
    public void postRemove(Post post) {
        // Update Redis cache
        log.info("postRemove");
        postRedisService.clear();
    }
    
}

File: src/main/java/com/lcaohoanq/nocket/domain/post/PostListener.kt
package com.lcaohoanq.nocket.domain.post

import com.lcaohoanq.nocket.domain.cache.IPostRedisService
import jakarta.persistence.*
import lombok.RequiredArgsConstructor
import lombok.extern.slf4j.Slf4j
import mu.KotlinLogging
import org.springframework.stereotype.Component

@Slf4j
@Component
class PostListener(
    private val postRedisService: IPostRedisService
) {

    private val logger = KotlinLogging.logger {}

    @PrePersist
    fun prePersist(post: Post?) {
        logger.info("prePersist")
    }

    @PostPersist //save = persis
    fun postPersist(post: Post?) {
        // Update Redis cache
        logger.info("postPersist")
        postRedisService.clear()
    }

    @PreUpdate
    fun preUpdate(post: Post?) {
        //ApplicationEventPublisher.instance().publishEvent(event);
        logger.info("preUpdate")
    }

    @PostUpdate
    fun postUpdate(post: Post?) {
        // Update Redis cache
        logger.info("postUpdate")
        postRedisService.clear()
    }

    @PreRemove
    fun preRemove(post: Post?) {
        //ApplicationEventPublisher.instance().publishEvent(event);
        logger.info("preRemove")
    }

    @PostRemove
    fun postRemove(post: Post?) {
        // Update Redis cache
        logger.info("postRemove")
        postRedisService.clear()
    }
}
